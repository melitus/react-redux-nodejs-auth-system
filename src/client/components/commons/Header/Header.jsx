//@flow
import React from 'react';
import type { Node as ReactNode} from 'react';

export type HeaderProps = {
  title: string,
};
const Header = ({ title } : HeaderProps): ReactNode =>  (
  <h1>{title}</h1>
);

Header.defaultProps = {
  title: 'Header Component'
};


export default Header;
