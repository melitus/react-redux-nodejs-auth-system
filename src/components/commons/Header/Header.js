import React from 'react';
import { Link } from 'react-router-dom';

const links = [
  {
    to: '/',
    text: 'Homepage'
  },
  {
    to: '/about',
    text: 'About'
  },
  {
    to: '/profile/1',
    text: 'Profile 1'
  },
  {
    to: '/profile/2',
    text: 'Profile 2'
  },
  {
    to: '/login',
    text: 'Login',
    auth: false
  },
  {
    to: '/dashboard',
    text: 'Dashboard',
    auth: true
  },
  {
    to: '/logout',
    text: 'Logout',
    auth: true
  },
  {
    to: '/this-is-broken',
    text: 'Broken Page'
  }
];

const isCurrent = (to, current) => {
  if (to === '/' && current === to) {
    return true;
  } else if (to !== '/' && current.includes(to)) {
    return true;
  }

  return false;
};

const HeaderLink = ({ to, text, current }) => (
  <li className={isCurrent(to, current) ? 'current' : ''}>
    <Link to={to}>{text}</Link>
  </li>
);

export default ({ isAuthenticated, current }) => (
  <header id="header">
    <h1 id="title">Awesome Login and Registration portal</h1>
    <ul id="links">
      {links.map((link, index) => {
        const TheLink = <HeaderLink key={index} current={current} {...link} />;

        if (link.hasOwnProperty('auth')) {
          if (link.auth && isAuthenticated) {
            return TheLink;
          } else if (!link.auth && !isAuthenticated) {
            return TheLink;
          }

          return null;
        }

        return TheLink;
      })}
    </ul>
  </header>
);
