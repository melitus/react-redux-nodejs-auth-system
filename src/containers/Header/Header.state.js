import { connect } from 'react-redux';
import Header from '../../components/commons/Header/Header';

export const mapDispatchToProps = {};

export const mapStateToProps = () => ({});

export default connect(mapStateToProps, mapDispatchToProps)(Header);
