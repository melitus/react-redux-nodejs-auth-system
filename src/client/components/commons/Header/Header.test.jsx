import React from 'react';
import enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15.4';

import Header from './Header';

enzyme.configure({ adapter: new Adapter() });


describe('<Header />', () => {
  it('should render Footer', () => {
    const wrapper = shallow(
      <Header />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
