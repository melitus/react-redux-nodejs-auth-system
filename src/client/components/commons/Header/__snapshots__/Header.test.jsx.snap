exports[`<Header /> should render Footer 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    title="Header Component" />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Header Component",
    },
    "ref": null,
    "rendered": "Header Component",
    "type": "h1",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Header Component",
      },
      "ref": null,
      "rendered": "Header Component",
      "type": "h1",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenFourAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
